internal/authtoken/repository.go

```diff
@@ -30,6 +29,11 @@ type Repository struct {
 	kms    *kms.Kms
 	// defaultLimit provides a default for limiting the number of results returned from the repo
 	defaultLimit int
+
+	// defaultTimeToLive and defaultTimeToStale durations ensure we do not accidentally set TTLs
+	// of zero value
+	timeToLiveDuration  time.Duration
```

Comment:
Don't add these here -- any function that needs to make use of this should instead take in options, which would provide the defaults if nothing is explicitly set.

Reply:
@talanknight and I went back and forth on this a bit. The reason we chose this route was that TTL isn't optional, it's required. So in the end, we decided to set it on the repo the same way we do for level because we couldn't think of a reason you'd ever want a 0 TTL.

Reply:
You don't need to store those in the repository itself --  you only need to override the local package var if the value is set. That's in theory racy but in practice you should only be set at startup, and only if the option coming in is non-zero. That allows you to remove a bunch of this boilerplate here and in tests.

Reply:
cleaned this up